### PR TITLE
add VCPKG_POST_PORTFILE_INCLUDES

### PR DESCRIPTION
**Pull request by Neumann-A**
docs pr: https://github.com/microsoft/vcpkg-docs/pull/336
vcpkg pr: https://github.com/microsoft/vcpkg/pull/25847

### CODE REVIEW

File: src/vcpkg/commands.build.cpp [hunk 1]
```diff
@@ -774,6 +776,16 @@ namespace vcpkg
...
         }
 
+        auto& post_portfile_includes = action.pre_build_info(VCPKG_LINE_INFO).post_portfile_includes;
+        std::string all_post_portfile_includes;
+        if (!post_portfile_includes.empty())
+            all_post_portfile_includes = std::accumulate(
+                std::next(post_portfile_includes.begin()),
+                post_portfile_includes.end(),
+                post_portfile_includes[0].generic_u8string(),
+                [](const Path& a, const Path& b) { return a.generic_u8string() + ";" + b.generic_u8string(); });
```
Comment (Neumann-A) [inline src/vcpkg/commands.build.cpp:786]: there is probably a besser way to do this with the helpers vcpkg has but I am mostly unfamiliar with them.

File: src/vcpkg/commands.build.cpp [hunk 2]
```diff
@@ -1220,6 +1232,18 @@ namespace vcpkg
                 Hash::get_file_hash(fs, file, Hash::Algorithm::Sha256).value_or_exit(VCPKG_LINE_INFO));
         }
 
+        for (size_t i = 0; i < abi_info.pre_build_info->post_portfile_includes.size(); ++i)
+        {
+            auto& file = abi_info.pre_build_info->post_portfile_includes[i];
+            if (file.is_relative() || !fs.is_regular_file(file) || file.extension() != ".cmake")
```
Comment (Neumann-A) [inline src/vcpkg/commands.build.cpp:1238]: should the extensions check be dropped? CMake can include arbitrary files and if it is not in cmake syntax, cmake will complain about it.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you!

**Comment by BillyONeal** [comment]
Sorry, I normally wouldn't push even nitpick fixes like that into your PR unless I was going to merge the thing; just as I was doing that I noticed the comment in the other PR ( https://github.com/microsoft/vcpkg/pull/25847#pullrequestreview-2104003043 )

This looks good to me and I asked Victor to OK it.

**Comment by Neumann-A** [comment]
@BillyONeal I don't mind. If I would mind I could disable the checkbox allowing you to do that.

**Review by vicroms — APPROVED** [review]
LGTM

**Pull request merged**